modules/template/controllers/ContainerContentController.php

```diff
@@ -261,4 +261,45 @@ public function actionMoveItem($elementContentId, $itemId, $step)
         ]);
     }
 
+    public function actionExportInstance()
+    {
+        $model = TemplateInstance::findOne(['id' => Yii::$app->request->get('id')]);
```

Comment:
Do we need some Access checking here?

Reply:
@luke- This controller has the behavior [TemplateAccessFilter](https://github.com/humhub/custom-pages/blob/enh/61-export-import-template-instance/modules/template/components/TemplateAccessFilter.php#L21-L37), it is restricted for non admins.

Reply:
@yurabakhtin Yes, this is just checking whether I am a Space admin for the Space. However, I could pass on the ID of another Space. Or am I misunderstanding something here?

Reply:
@luke- You are right, I have implemented the additional checking by new method `isValidTemplateInstance()` [here](https://github.com/humhub/custom-pages/pull/407/commits/767203e94a9a481917c49da8527f42c58268c909#diff-ddfc119f7ad895c793c0260c7fdc87730c2c5705b348ee51db57df4321a6184fR351).
Now it works this way:
- Check if the current user is global admin or space admin - [here](https://github.com/humhub/custom-pages/pull/407/commits/767203e94a9a481917c49da8527f42c58268c909#diff-666f2e3863b9cf2169f854e7dc815c8ebeac226e6105e6f7c09e86d9dfa721e2R26-R32)
- Then check the added/edited/moved/deleted element is called from correct Space or from Global.